### PR TITLE
Added new deploy method in sources

### PR DESCRIPTION
**Pull request by jcortes**
### <samp>🤖 Generated by Copilot at d7c7042</samp>

This pull request improves various sources and actions for Google Drive and Google Docs, adding new features, fixing bugs, and refactoring code. It also updates the versions of the sources, actions, and packages to reflect the changes.

### CODE REVIEW

File: components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs [hunk 1]
```diff
@@ -54,6 +56,20 @@ export default {
       },
     },
   },
+  hooks: {
+    async deploy() {
+      console.log("Deploying...");
```
Comment (dylburger) [inline components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]:
```suggestion
      console.log("Deploying");
```

File: components/google_drive/sources/new-files-instant/new-files-instant.mjs [hunk 2]
```diff
@@ -40,6 +40,17 @@ export default {
     },
   },
   hooks: {
+    async deploy() {
+      console.log("Deploying...");
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs [hunk 3]
```diff
@@ -17,12 +17,27 @@ export default {
...
     "Emits a new event any time a file comment is added, modified, or deleted in your linked Google Drive",
-  version: "0.1.0",
+  version: "0.1.1",
   type: "source",
   // Dedupe events based on the "x-goog-message-number" header for the target channel:
   // https://developers.google.com/drive/api/v3/push#making-watch-requests
   dedupe: "unique",
   hooks: {
+    async deploy() {
+      console.log("Deploying...");
```
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs [hunk 4]
```diff
@@ -20,11 +20,25 @@ export default {
...
   description: "Emit new event any time any file in your linked Google Drive is added, modified, or deleted",
-  version: "0.1.0",
+  version: "0.1.1",
   type: "source",
   // Dedupe events based on the "x-goog-message-number" header for the target channel:
   // https://developers.google.com/drive/api/v3/push#making-watch-requests
   dedupe: "unique",
+  hooks: {
+    async deploy() {
+      console.log("Deploying...");
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs [hunk 5]
```diff
@@ -20,11 +20,25 @@ export default {
...
   description: "Emit new event any time any folder in your linked Google Drive is added, modified, or deleted",
-  version: "0.1.0",
+  version: "0.1.1",
   type: "source",
   // Dedupe events based on the "x-goog-message-number" header for the target channel:
   // https://developers.google.com/drive/api/v3/push#making-watch-requests
   dedupe: "unique",
+  hooks: {
+    async deploy() {
+      console.log("Deploying...");
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-files-instant/new-files-instant.mjs (hunk 2 above)
Comment (dylburger) [inline components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs (hunk 3 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs (hunk 4 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs (hunk 5 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]: Changed to use active voice.

File: components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs (hunk 1 above)
Comment (dylburger) [inline components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-files-instant/new-files-instant.mjs (hunk 2 above)
Comment (dylburger) [inline components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs (hunk 3 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs (hunk 4 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs (hunk 5 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]:
```suggestion
      console.log("Deploying");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]: Removed ellipsis to follow Google's Developer docs style guide

File: components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylburger) [reply components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]: Changed to console.log("Deploying..."); to follow Google's Developer docs style guide.

File: components/google_drive/sources/new-files-instant/new-files-instant.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dylburger) [reply components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]: Changed to console.log("Deploying..."); to follow Google's Developer docs style guide.

File: components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs (hunk 3 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]:
```suggestion
      console.log("Deploying...");
```

File: components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs (hunk 4 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]: Changed to console.log("Deploying..."); to follow Google's Developer docs style guide.

File: components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs (hunk 5 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]: Changed to console.log("Deploying..."); to follow Google's Developer docs style guide.

File: components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs (hunk 1 above)
Comment (dylburger) [inline components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/changes-to-specific-files-shared-drive/changes-to-specific-files-shared-drive.mjs:61]: Changed to use active voice.

File: components/google_drive/sources/new-files-instant/new-files-instant.mjs (hunk 2 above)
Comment (dylburger) [inline components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-files-instant/new-files-instant.mjs:44]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs (hunk 3 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.mjs:27]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs (hunk 4 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]:
```suggestion
      console.log("Deploying...");
```

Comment (dylburger) [reply components/google_drive/sources/new-or-modified-files/new-or-modified-files.mjs:30]: Changed to use active voice.

File: components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs (hunk 5 above)
Comment (dylburger) [inline components/google_drive/sources/new-or-modified-folders/new-or-modified-folders.mjs:30]:
```suggestion
      console.log("Deploying...");
```

File: components/google_docs/actions/append-image/append-image.mjs [hunk 6]
```diff
@@ -4,7 +4,7 @@ export default {
   key: "google_docs-append-image",
   name: "Append Image to Document",
   description: "Appends an image to the end of a document. [See the docs](https://developers.google.com/docs/api/reference/rest/v1/documents/request#InsertInlineImageRequest)",
-  version: "0.0.2",
+  version: "0.0.1",
```
Comment (luancazarine) [inline components/google_docs/actions/append-image/append-image.mjs:7]:
```suggestion
  version: "0.0.3",
```

Comment (luancazarine) [inline components/google_docs/actions/append-image/append-image.mjs:7]:
```suggestion
  version: "0.0.3",
```

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @jcortes, thanks for your contribution! I've just left a minor comment.

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Google_Drive_sources_6490_1324.pdf](https://github.com/PipedreamHQ/pipedream/files/11484637/Google_Drive_sources_6490_1324.pdf)

**Comment by vunguyenhung** [comment]
Hi @jcortes, I have tested the following sources that you have changed:
- changes-to-specific-files-shared-drive
- new-or-modified-comments
- new-files-instant
- new-or-modified-files
- new-or-modified-folders

Though, the 2 sources still failed:
- changes-to-specific-files-shared-drive
- new-or-modified-comments

Would you mind checking in the test report below?
[Google_Drive_sources_6490_1057.pdf](https://github.com/PipedreamHQ/pipedream/files/11504212/Google_Drive_sources_6490_1057.pdf)

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Google_Drive_sources_6490_0958.pdf](https://github.com/PipedreamHQ/pipedream/files/11538816/Google_Drive_sources_6490_0958.pdf)

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Google_Drive_sources_6490_0740.pdf](https://github.com/PipedreamHQ/pipedream/files/11549400/Google_Drive_sources_6490_0740.pdf)

**Review by luancazarine — APPROVED** [review]

**Comment by trialsin** [comment]
I am having issues with the step **upsert_row**. The other **add_single_row** and **update_row** are okay. I checked the parameters and it seems to be correct.

Apologies for posting here, but I couldn't resolve in other channels.

Here is another person with the same problem: [https://pipedream.com/community/t/upsert-row-on-google-sheets-throw-err-unable-to-parse-range/1979](https://pipedream.com/community/t/upsert-row-on-google-sheets-throw-err-unable-to-parse-range/1979)


```
Error
Unable to parse range: Plan!#ERROR!:#ERROR!

DETAILS
    at Gaxios._request (/tmp/__pdg__/dist/code/021035a6bb454b1c3cef29209abbac28b16fb0268d29e88192565ce11e11286a/node_modules/.pnpm/gaxios@5.1.3/node_modules/gaxios/build/src/gaxios.js:140:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at OAuth2Client.requestAsync (/tmp/__pdg__/dist/code/021035a6bb454b1c3cef29209abbac28b16fb0268d29e88192565ce11e11286a/node_modules/.pnpm/google-auth-library@8.9.0/node_modules/google-auth-library/build/src/auth/oauth2client.js:382:18)
    at Object.updateSpreadsheet (file:///tmp/__pdg__/dist/code/021035a6bb454b1c3cef29209abbac28b16fb0268d29e88192565ce11e11286a/code/google_sheets/google_sheets.app.mjs:332:15)
    at Object.updateRow (file:///tmp/__pdg__/dist/code/021035a6bb454b1c3cef29209abbac28b16fb0268d29e88192565ce11e11286a/code/google_sheets/google_sheets.app.mjs:491:14)
    at Function.all (null:null:null)
    at Object.run (file:///tmp/__pdg__/dist/code/021035a6bb454b1c3cef29209abbac28b16fb0268d29e88192565ce11e11286a/code/google_sheets/actions/upsert-row/upsert-row.mjs:181:9)
    at null.executeComponent (/var/task/launch_worker.js:229:22)
    at MessagePort.messageHandler (/var/task/launch_worker.js:726:28)

```

image

**Comment by dannyroosevelt** [comment]
@trialsin I just responded to you in Slack

**Pull request merged**